### PR TITLE
Added omitempty for SecurityConfig type

### PR DESCRIPTION
**Pull request by BLasan**
## Purpose
Applying petstorejwt secret will get failed due to the missing parameter "omitempty" in some non-essential fields of type SecurityConfig

## Goals
Add "omitempty" for those non-essential fields of type SecurityConfig

## Related Issue
#226

### CODE REVIEW

**Comment by BLasan** [comment]
@renuka-fernando @pubudu538 please review

File: api-operator/pkg/apis/wso2/v1alpha1/security_types.go [hunk 1]
```diff
@@ -66,14 +66,14 @@ type SecurityList struct {
...
+	Alias                string `json:"alias,omitempty"`
+	Endpoint             string `json:"endpoint,omitempty"`
+	Credentials          string `json:"credentials,omitempty"`
 	Issuer               string `json:"issuer"`
-	Audience             string `json:"audience"`
+	Audience             string `json:"audience,omitempty"`
 	ValidateSubscription bool   `json:"validateSubscription,omitempty"`
 	ValidateAllowedAPIs  bool   `json:"validateAllowedAPIs,omitempty"`
-	JwksURL              string `json:"jwksURL"`
+	JwksURL              string `json:"jwksURL,omitempty"`
```
Comment (renuka-fernando) [inline api-operator/pkg/apis/wso2/v1alpha1/security_types.go:76]: Shall we make fields `Certificate` and `Issuer` are also optional

- In security type "basic" only required "credentials"
```yaml
apiVersion: wso2.com/v1alpha1
kind: Security
metadata:
  name: petstorebasic
spec:
  type: basic
  securityConfig:
    - credentials: secret-basic
```

Other types
- Oauth
```yaml
apiVersion: wso2.com/v1alpha1
kind: Security
metadata:
  name: petstoreoauth
spec:
  type: Oauth
  securityConfig:
    - certificate: wso2am320-secret #create secret with certificate and add secret name
      endpoint: "https://wso2apim.wso2:9443"
      credentials: oauth-credentials
```

- JWT
```yaml
apiVersion: wso2.com/v1alpha1
kind: Security
metadata:
  name: petstorejwt
spec:
  type: JWT
  securityConfig:
    - issuer: https://wso2apim:32001/oauth2/token
      #create secret with certificate and add secret name
      certificate: wso2am320-secret
      validateSubscription: false
```

**Comment by BLasan** [comment]
@renuka-fernando Made the requested changes. Please review

**Comment by pubudu538** [comment]
Let's not do these changes in the patch version. So I am closing this PR.

**Pull request closed without merging**